client/context/viper.go

```diff
@@ -15,13 +20,48 @@ func NewCoreContextFromViper() core.CoreContext {
 	if nodeURI != "" {
 		rpc = rpcclient.NewHTTP(nodeURI, "/websocket")
 	}
+	chainID := viper.GetString(client.FlagChainID)
+	// if chain ID is not specified manually, read chain ID from genesis file if present
+	if chainID == "" {
+		cfg, err := tcmd.ParseConfig()
+		if err == nil {
+			genesisFile := cfg.GenesisFile()
+			bz, err := ioutil.ReadFile(genesisFile)
+			if err == nil {
+				var doc tmtypes.GenesisDoc
+				err = json.Unmarshal(bz, &doc)
+				if err == nil {
```

Comment:
Bad practice to nest if statements like this. I understand that we don't return an error in this function so you are nesting the operations but this suggests to me that this nest should be a seperate function which does return an error.

Reply:
Good point, updated.